index.js

```diff
@@ -4,9 +4,11 @@ import inherits from 'inherits'
  * Create a custom constructor which can be modified without affecting the
  * original class.
  *
- * @template {Function} Class
+ * @template Instance
+ * @template Args
+ * @template {{ new(...Args): Instance }} Class
```

Comment:
These reflect the arguments that the constructor accepts and the instance type it returns